File: server/models/mission.js
"use strict";

class Mission {

    constructor (id, name, description, launch, type, wikiURL, infoURLs, agencies, events) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.launch = launch;
        this.type = type;
        this.wikiURL = wikiURL;
        this.infoURLs = infoURLs;
        this.agencies = agencies;
        this.events = events;
    }

    static fromJSON (json) {
        let x = JSON.parse(json);
        return new Mission(x.id, x.name, x.description, x.launch, x.type, x.wikiURL, x.infoURLs, x.agencies, x.events)
    }
}

File: src/main/java/com/eccff/launchbot/models/Mission.java
package com.eccff.launchbot.models;

import lombok.Data;

import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import java.util.List;

/**
 * https://launchlibrary.net/1.2/mission
 */

@Data
@Entity
public class Mission {

    public Mission() {}

    @Id
    private int id;
    private String name;
    private String description;
    @OneToOne
    private Launch launch;
    private int type;
    private String wikiURL;
    @ElementCollection(targetClass=String.class)
    private List<String> infoURLs;
    @ElementCollection(targetClass=Agency.class)
    private List<Agency> agencies;
    @ElementCollection(targetClass=MissionEvent.class)
    private List<MissionEvent> events;

}
